### PR TITLE
:bug:  Add readiness and fix startup

### PR DESCRIPTION
**Pull request by mjudeikis**
## Summary

Moves back to fork to fix remote conditions.
Add simple service readiness to print if we are not ready preventing "handover to user" if its not ready.

## Related issue(s)

Fixes https://github.com/kcp-dev/generic-controlplane/issues/29

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by mjudeikis** [comment]
Will remove fork too now as have fix

File: server/cmd/server.go [hunk 1]
```diff
@@ -210,7 +211,24 @@ func Run(ctx context.Context, opts options.CompletedOptions) error {
...
+
+	go func() {
+		if err := prepared.Run(ctx); err != nil {
+			klog.Fatal(err, "Failed to run server")
+		}
+	}()
+
+	// wait for the server to be ready
+	klog.Info("Waiting for control plane to be ready")
+	err = readiness.WaitForReady(ctx, completed.Options.AdminAuthentication.KubeConfigPath)
```
Comment (sttts) [inline server/cmd/server.go:224]: is this just for cosmetics, to have nice output?

Comment (mjudeikis) [reply server/cmd/server.go:224]: Right, to log out when something is not yet ready, but in reality control-plane is semi-functional already.

Comment (mjudeikis) [reply server/cmd/server.go:224]: This was my mistake. Perception was that its "healthy" but something was not starting right.

**Pull request merged**